src/utils.c

```diff
@@ -269,20 +269,20 @@ static avifBool avifDoubleToUnsignedFractionImpl(double v, uint32_t maxNumerator
     return AVIF_TRUE;
 }
 
-avifBool avifDoubleToSignedFraction(double v, int32_t * numerator, uint32_t * denominator)
+avifBool avifDoubleToSignedFraction(double v, avifSignedFraction * fraction)
 {
     uint32_t positive_numerator;
-    if (!avifDoubleToUnsignedFractionImpl(fabs(v), INT32_MAX, &positive_numerator, denominator)) {
+    if (!avifDoubleToUnsignedFractionImpl(fabs(v), INT32_MAX, &positive_numerator, &fraction->d)) {
```

Comment:
In `avifDoubleToUnsignedFractionImpl()`, the variable `maxD` is declared as `uint64_t`, but it is only used in a comparison with `newD`, which is a `double`. So `maxD` could be declared as `double`. If `maxD` should have an integer type, `uint32_t` is sufficient.